core/controllers/collection_editor.py

```diff
@@ -102,20 +104,40 @@ def get(self, collection_id):
         collection = collection_services.get_collection_by_id(
             collection_id, strict=False)
 
+        can_edit = (
+            bool(self.user_id) and
+            self.username not in config_domain.BANNED_USERNAMES.value and
+            rights_manager.Actor(self.user_id).can_edit(
+                feconf.ACTIVITY_TYPE_COLLECTION, collection_id))
+
+
+        user_permissions = rights_manager.get_collection_rights(
+            collection_id).to_dict()
+        owner_names = user_permissions['owner_names']
+        editor_names = user_permissions['editor_names']
+        viewer_names = user_permissions['viewer_names']
+
         self.values.update({
-            'can_edit': True,
+            'can_edit': can_edit,
+            'can_modify_roles': rights_manager.Actor(
+                self.user_id).can_modify_roles(
+                    feconf.ACTIVITY_TYPE_COLLECTION, collection_id),
             'can_unpublish': rights_manager.Actor(
                 self.user_id).can_unpublish(
                     feconf.ACTIVITY_TYPE_COLLECTION, collection_id),
             'collection_id': collection.id,
+            'collection_version': collection.version,
+            'editors': editor_names,
```

Comment:
editors --> editor_names (this disambiguates more clearly between usernames and ids). Ditto for others.